src/System.Data.SqlClient/src/Interop/LocaleMapperApiHelper.cs

```diff
@@ -0,0 +1,482 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
It seems it will be more make sense not to use this interop.

I will prefer to move this inside Locale folder instead as only windows are using interop call to Windows API.

and we will rename LocaleInterop.cs as something else.

btw, what do you mean by Unix Specific folder ? 
This file will be compile on unix only. it is defined in project file.
And it seems other library also put unix and windows implementation on the same folder.
e.g System.Text.Encoding.CodePages. 
This will be easier to see different implementation on the same folder.

And also the reason why I don't separate LocaleInterop.Windows.cs or LocaleInterop.Unix.cs, is because it will have duplicate code inside there. 
So for minimal change and cleaner way, I use Interface to separate this implementation and switch this helper using defined constant for windows / Non windows.

This helper will be compile as it needs. e.g MapperApi will be compiled on Unix only and windowsApi will be compiled on windows only.